gratipay/models/participant/email.py

```diff
@@ -280,11 +281,19 @@ def save_email_address(self, cursor, address):
     def finish_package_claims(self, cursor, nonce, *packages):
         """Create teams if needed and associate them with the packages.
         """
-        cursor.run('DELETE FROM claims WHERE nonce=%s', (nonce,))
-        package_ids = []
+        if not packages:
+            raise NoPackages()
+
+        package_ids, teams, team_ids = [], [], []
         for package in packages:
-            package.ensure_team(cursor, self)
             package_ids.append(package.id)
+            team = package.get_or_create_linked_team(cursor, self)
+            teams.append(team)
+            team_ids.append(team.id)
+        review_url = self.app.project_review_repo.create_issue(*teams)
+
+        cursor.run('DELETE FROM claims WHERE nonce=%s', (nonce,))
+        cursor.run('UPDATE teams SET review_url=%s WHERE id=ANY(%s)', (review_url, team_ids,))
```

Comment:
If you're seeing `cursor` instead of `self.db` then yes, the expectation is that this is happening in a transaction. The transaction begins when the cursor is [created](https://github.com/gratipay/gratipay.com/blob/ff933e994eae0ca5e1124bbc4e3b7ea824653dfd/gratipay/models/participant/email.py#L233), which happens in a context manager invocation. It will be rolled back if there's an exception, or committed when the context manager exits normally. [Docs](http://postgres-py.readthedocs.io/en/latest/#context-managers).

In terms of a test, I believe we're covered by the one added in eaa8699912e8571f0d6750955672795d3fbf7ff5 under #4397, no?